core/src/main/scala/org/apache/spark/util/ArrayReflect.java

```diff
@@ -0,0 +1,357 @@
+/*
```

Comment:
Given its provenance I'm not sure we can say it's licensed to the ASF, but can state its source and apparent license here.